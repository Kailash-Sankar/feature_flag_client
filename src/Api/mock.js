function getPackages() {
  return {
    "0": { id: 0, name: "Custom" },
    "1": { id: 1, name: "Silver" },
    "2": { id: 2, name: "Gold" },
    "3": { id: 3, name: "Diamond" }
  };
}

// hardcoded for demo
function getPackageFeatures() {
  return {
    "0": [],
    "1": {
      ad_slates: {
        id: "ad_slates",
        status: 2
      }
    },
    "2": [],
    "3": []
  };
}

export default {
  getPackages,
  getPackageFeatures
};
