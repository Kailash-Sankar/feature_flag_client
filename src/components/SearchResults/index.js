import React from "react";
import { Collapse, List } from "antd";
const { Panel } = Collapse;

const statusMap = {
  "0": "Disabled",
  "1": "Internal",
  "2": "All"
};

export function RenderList({ data }) {
  return (
    <List
      itemLayout="horizontal"
      dataSource={data}
      renderItem={item => (
        <List.Item extra={statusMap[item.status]}>
          <List.Item.Meta title={item.name} description={item.description} />
        </List.Item>
      )}
    />
  );
}

function SearchResults({ result }) {
  console.log("SR", result);
  return (
    <div>
      <Collapse bordered={false} defaultActiveKey={[]}>
        {result.map(r => {
          return (
            <Panel header={r.name} extra={""} key={r.id}>
              <RenderList data={r.features} />
            </Panel>
          );
        })}
      </Collapse>
    </div>
  );
}

export default SearchResults;
