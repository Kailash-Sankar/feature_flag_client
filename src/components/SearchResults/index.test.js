import { shallow } from "enzyme";
import React from "react";
import mock from "@mock/data";
import SearchResults from "./index";

test("Search Results", () => {
  const wrapper = shallow(<SearchResults result={mock.features} />);
  expect(wrapper).toMatchSnapshot();
});
