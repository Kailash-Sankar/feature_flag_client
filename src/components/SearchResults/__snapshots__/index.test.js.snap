// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results 1`] = `
<div>
  <Collapse
    bordered={false}
    defaultActiveKey={Array []}
    expandIconPosition="left"
  >
    <CollapsePanel
      extra={
        <span
          style={
            Object {
              "color": "#757575",
              "fontSize": 12,
            }
          }
        >
          Last Update:Invalid Date
        </span>
      }
      header="Feature One"
      key="ff_one"
    >
      <RenderList />
    </CollapsePanel>
    <CollapsePanel
      extra={
        <span
          style={
            Object {
              "color": "#757575",
              "fontSize": 12,
            }
          }
        >
          Last Update:Invalid Date
        </span>
      }
      header="Feature Two"
      key="ff_two"
    >
      <RenderList />
    </CollapsePanel>
    <CollapsePanel
      extra={
        <span
          style={
            Object {
              "color": "#757575",
              "fontSize": 12,
            }
          }
        >
          Last Update:Invalid Date
        </span>
      }
      header="Feature Three"
      key="ff_three"
    >
      <RenderList />
    </CollapsePanel>
  </Collapse>
</div>
`;
